Code/Framework/AzToolsFramework/AzToolsFramework/API/PythonLoader.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+
+#include <AzToolsFramework/API/PythonLoader.h>
+#include <AzToolsFramework_Traits_Platform.h>
+#include <AzCore/IO/FileIO.h>
+#include <AzCore/IO/GenericStreams.h>
+#include <AzCore/IO/Path/Path.h>
+#include <AzCore/Math/Sha1.h>
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/string/string.h>
+#include <AzCore/std/string/string_view.h>
+#include <AzCore/std/string/conversions.h>
+#include <AzCore/std/string/tokenize.h>
+#include <AzCore/Settings/ConfigParser.h>
+#include <AzCore/Utils/Utils.h>
+
+namespace AzToolsFramework::EmbeddedPython
+{
+    PythonLoader::PythonLoader()
+    {
+        LoadRequiredModules();
+    }
+
+    PythonLoader::~PythonLoader()
+    {
+        UnloadRequiredModules();
+    }
+
+    AZ::IO::FixedMaxPath PythonLoader::GetDefault3rdPartyPath(bool createOnDemand)
+    {
+        AZ::IO::FixedMaxPath thirdPartyEnvPathPath;
+        static constexpr const char* env3rdPartyKey = "LY_3RDPARTY_PATH";
+
+        char env3rdPartyPath[AZ::IO::MaxPathLength] = {'\0'};
+        auto envOutcome = AZ::Utils::GetEnv(AZStd::span(env3rdPartyPath), env3rdPartyKey);
+        if (envOutcome && (strlen(env3rdPartyPath) > 0))
+        {
+            // If so, then use the path that is set as the third party path
+            thirdPartyEnvPathPath = AZ::IO::FixedMaxPath(env3rdPartyPath).LexicallyNormal();
+        }
+        else
+        {
+            auto manifestPath = AZ::Utils::GetO3deManifestDirectory();
+            thirdPartyEnvPathPath = AZ::IO::FixedMaxPath(manifestPath) / "3rdParty";
+        }
+
+        AZStd::string thirdPartyPathString = thirdPartyEnvPathPath.String();
+        if ((!AZ::IO::SystemFile::IsDirectory(thirdPartyPathString.c_str())) && createOnDemand)
+        {
+            auto createPathResult = AZ::IO::SystemFile::CreateDir(thirdPartyPathString.c_str());
+            AZ_Assert(createPathResult, "Unable to create missing 3rd Party Folder '%s'", thirdPartyPathString.c_str())
+        }
+        return thirdPartyEnvPathPath;
+    }
```

Comment:
Well there is still a problem here and in that if a user uses the CMake cache variable option of `-DLY_3RDPARTY_PATH=<3rdPartyPath>` when configuring, then this logic will not pickup that path.

I know the Project Manager allows the user to customizer their 3rdParty directory in the [Engine Settings widget](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/EngineSettingsScreen.cpp#L69).

Perhaps this function should have a parameter for accepting a 3rdParty path supplied via the function argument and then check to see if that path exist before falling back to looking at the `<o3de-manifest>/3rdParty` directory.

Psuedocode wise.
```c++
       if (envOutcome && (strlen(env3rdPartyPath) > 0))
        {
            // If so, then use the path that is set as the third party path
            thirdPartyEnvPathPath = AZ::IO::FixedMaxPath(env3rdPartyPath).LexicallyNormal();
        }
        else if (AZ::IO::SystemFile::IsDirectory(userThirdPartyPath))
        {
            return userThirdPartyPath;
        }
        else
        {
            auto manifestPath = AZ::Utils::GetO3deManifestDirectory();
            thirdPartyEnvPathPath = AZ::IO::FixedMaxPath(manifestPath) / "3rdParty";
        }
```

Reply:
Well, this was meant to get the default 3rd party path. It can add an `#if defined(LY_3RDPARTY_PATH)` as a fallback (or primary)

Reply:
Created a new [Get3rdPartyDirectory](https://github.com/spham-amzn/o3de/blob/37bf736e0f325802f3059f916795cd624ba15280/Code/Framework/AzCore/AzCore/Utils/Utils.cpp#L415-L444) method to read the path from the manifest.json if possible, and added that to the logic